build.gradle

```diff
@@ -1,81 +1,95 @@
-apply plugin: 'java'
+plugins {
+	id "java"
+	id "com.github.jk1.dependency-license-report" version "0.3.11"
+}
 
 repositories {
+	mavenLocal()
 	jcenter()
+	mavenCentral()
 	maven { url "http://developer.marklogic.com/maven2" }
 }
 
-configurations {
-	all {
-		exclude module: 'commons-logging'
-		exclude module: 'log4j'
-	}
-}
-
 dependencies {
 	// Core dependencies needed for using Spring Batch, Spring JDBC, and marklogic-spring-batch
-	compile "com.marklogic:marklogic-spring-batch-core:0.11.0"
-	compile "com.marklogic:ml-javaclient-util:2.12.0"
-	compile "org.springframework:spring-jdbc:4.3.5.RELEASE"
+	compile "com.marklogic:marklogic-spring-batch-core:1.4.0"
+	compile "com.marklogic:spring-batch-rdbms:1.4.0"
 
 	// Using H2 as a sample database; can pull its JDBC driver from a Maven repository
-	runtime "com.h2database:h2:1.4.193"
+	testRuntime "com.h2database:h2:1.4.193"
 
 	// Can reference a local JDBC driver like this (Oracle, along with others, do not publish their JDBC driver to a public repo)
 	// runtime files("./path/to/ojdbc-6.jar")
 
-	// Forcing Spring to use logback instead of commons-logging
-	runtime "ch.qos.logback:logback-classic:1.1.8"
-	runtime group: "org.slf4j", name: "jcl-over-slf4j", version: "1.7.22"
-	runtime group: "org.slf4j", name: "slf4j-api", version: "1.7.22"
 }
 
-task migrate(type: JavaExec) {
+task migrateXml(type: JavaExec) {
```

Comment:
I liked the previous approach with a single "migrate" task, that seems simpler for the user. The Gradle task can figure out whether certain args should be included, right? Or if the arg doesn't have a value, shouldn't the program ignore it?

Reply:
The intent was really just to show different ways of migrating data using the same program.  I will change it back to a single migrate class.  

The main difference from the previous version is that the connection information to MarkLogic, the in-memory database, and the ML Job Repo are in property files.